File: app/models/ticket.rb
class Ticket < ApplicationRecord
  belongs_to :project
  belongs_to :assigned_dev, class_name: "User", inverse_of: :assigned_tickets, optional: true
  belongs_to :submitted_by, class_name: "User", inverse_of: :submitted_tickets, optional: true
  has_many :histories, dependent: :destroy
  has_many :comments, dependent: :destroy

  PRIORITIES = ["low", "medium", "high"] 
  STATUSES = ["new", "open", "in progress", "resolved", "additional information required"]
  TYPES = ["bug/errors", "feature requests", "comments", "training/document requests"]
  validates :title, :description, :project_id, :priority, :ticket_type, :status, presence: true
  validates :title, length: { maximum: 100 }
  validates :description, length: { maximum: 1000 }
  validates :priority, inclusion: { in: PRIORITIES, message: "Priority is not valid" }
  validates :status, inclusion: { in: STATUSES, message: "Status is not valid" }
  validates :ticket_type, inclusion: { in: TYPES, message: "Type is not valid" }

  # want project id unchangable
  attr_readonly :project_id
end

File: app/javascript/components/TicketsList/Ticket.js
import React from "react";

function Ticket({ title, priority, status, type, url }) {
  return (
    <tr>
      <td className="fs-4">
        <a href={url}>{title}</a>
      </td>
      <td className="fs-4">{priority}</td>
      <td className="fs-4">{status}</td>
      <td className="fs-4">{type}</td>
      <td className="fs-4">
        <a href={url}>View Ticket</a>
      </td>
    </tr>
  );
}

export default Ticket;
